Makefile

```diff
@@ -25,3 +25,23 @@ coverage:
 coveralls: coverage
 	go get github.com/mattn/goveralls
 	goveralls -coverprofile=$(COVERAGE_FILE) -service=github
+
+.PHONY: bench
+bench:
+	go clean -testcache
+	go test -bench=. -benchmem -benchtime=1s
```

Comment:
I suspect that 1 second is too little duration to get stable results in a benchmark with network interation. Can you run benchmarks with different durations/iterations count and estimate a noise? Also I would make this value configurable by an environment variable (and set a default in the makefile).

Reply:
Fixed. Added the variable `DURATION` for that goal.

Reply:
I have added the variable `DURATION` for controlling tests durations and the variable `COUNT` for ability to run each benchmark `${COUNT}` times and pass this result through benchstat, which gives us some statistic about all test runs. The output looks like this:
```
name                            time/op
ClientSerial-8                  71.3µs ± 8%
ClientSerialTyped-8             74.9µs ±22%
ClientFuture-8                   177µs ±19%
ClientFutureTyped-8              167µs ± 9%
ClientFutureParallel-8          24.3µs ± 8%
ClientFutureParallelTyped-8     24.4µs ± 7%
ClientParallel-8                26.4µs ±12%
ClientParallelMassive-8         51.5µs ±10%
ClientParallelMassiveUntyped-8  52.0µs ±13%
Connection_Replace-8            26.8µs ±12%
Connection_LargeSelect-8         102ms ±19%

name                            alloc/op
ClientSerial-8                    848B ± 0%
ClientSerialTyped-8             1.03kB ± 7%
ClientFuture-8                  65.2kB ± 0%
ClientFutureTyped-8             55.3kB ± 0%
ClientFutureParallel-8          1.09kB ±19%
ClientFutureParallelTyped-8     1.13kB ± 4%
ClientParallel-8                  978B ±13%
ClientParallelMassive-8         4.61kB ±12%
ClientParallelMassiveUntyped-8  2.85kB ±35%
Connection_Replace-8              961B ± 0%
Connection_LargeSelect-8         214MB ± 0%

name                            allocs/op
ClientSerial-8                    20.0 ± 0%
ClientSerialTyped-8               16.6 ± 4%
ClientFuture-8                   1.72k ± 0%
ClientFutureTyped-8              1.22k ± 0%
ClientFutureParallel-8            23.0 ± 0%
ClientFutureParallelTyped-8       21.2 ± 8%
ClientParallel-8                  22.4 ± 3%
ClientParallelMassive-8            100 ±10%
ClientParallelMassiveUntyped-8    40.8 ±25%
Connection_Replace-8              21.0 ± 0%
Connection_LargeSelect-8          700k ± 0%

```

AFAIU this percents means the biggest deviation of the value between all ${COUNT} outputs.